src/core/lib/resource_quota/memory_quota.cc

```diff
@@ -424,6 +433,21 @@ void BasicMemoryQuota::Take(size_t amount) {
   if (prior >= 0 && prior < static_cast<intptr_t>(amount)) {
     if (reclaimer_activity_ != nullptr) reclaimer_activity_->ForceWakeup();
   }
+
+  if (IsFreeLargeAllocatorEnabled()) {
+    GrpcMemoryAllocatorImpl* chosen_allocator = nullptr;
+    int shard_idx = rand() % big_allocators_.shards.size();
```

Comment:
We should avoid using `rand()`... it's terrible on several axis (concurrency, and randomness).
It's probably ok to put an `absl::Bitgen` as a member variable and use `absl::Uniform` here.

Reply:
Discussed offline - using atomic per allocator to decide shard.